packages/block-library/src/navigation/edit/edit-overlay-button.js

```diff
@@ -0,0 +1,146 @@
+import { parse, serialize } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+const DEFAULT_NAVIGATION_OVERLAY = {
+	slug: 'navigation-overlay',
+	content: {
+		raw: `<!-- wp:group {"layout":{"type":"flex","orientation":"vertical","justifyContent":"center"}} -->
+<div class="wp-block-group"><!-- wp:navigation {"layout":{"type":"flex","orientation":"vertical","justifyContent":"left"}} /--></div>
+<!-- /wp:group -->`,
+	},
+};
+
+export default function EditOverlayButton( { navRef } ) {
+	const [ navTitle ] = useEntityProp(
+		'postType',
+		'wp_navigation',
+		'title',
+		navRef
+	);
+
+	// get the template part with the slug navigation-overlay
+	const { baseOverlay, customOverlay } = useSelect(
+		( select ) => {
+			const themeSlug = select( coreStore ).getCurrentTheme()?.stylesheet;
+
+			// Try for overlay provided by Theme (falling back to default
+			// provided by Core).
+			const _baseOverlay = themeSlug
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'wp_template_part',
+						`${ themeSlug }//navigation-overlay`
+				  )
+				: null;
+
+			const _customOverlay = themeSlug
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'wp_template_part',
+						`${ themeSlug }//navigation-overlay-${ navRef }`
+				  )
+				: null;
+
+			return {
+				baseOverlay: _baseOverlay,
+				customOverlay: _customOverlay,
+			};
+		},
+		[ navRef ]
+	);
+
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	const history = useHistory();
+
+	function findNavigationBlock( blocks ) {
+		const block = blocks[ 0 ];
+		if ( block.name === 'core/navigation' ) {
+			return block;
+		}
+		if ( block.innerBlocks.length ) {
+			return findNavigationBlock( block.innerBlocks );
+		}
+		return null;
+	}
+
+	function goToOverlayEditor( overlayId ) {
+		history.push( {
+			postId: overlayId,
+			postType: 'wp_template_part',
+			canvas: 'edit',
+		} );
+	}
+
+	async function handleEditOverlay( event ) {
+		event.preventDefault();
+
+		// If there is already a Custom Overlay for this Navigation Menu
+		// the go to the editor for that overlay template part.
+		if ( customOverlay ) {
+			goToOverlayEditor( customOverlay.id );
+			return;
+		}
+
+		// No custom overoverlay - create one from base template.
+		// TODO: catch and handle errors.
+		const overlayBlocks = buildOverlayBlocks( baseOverlay, navRef );
+		const newOverlay = await createCustomOverlay( overlayBlocks );
+
+		goToOverlayEditor( newOverlay?.id );
+		return;
+	}
+
+	function buildOverlayBlocks( baseOverlay, navRef ) {
+		const parsedBlocks = parse( baseOverlay.content.raw );
+		const overlayNavBlock = findNavigationBlock( parsedBlocks );
+
+		// Update the Navigation block in the overlay to use
+		// the same ref as the parent block.
+		// Todo: what happens if ref doesn't exist?
+		// Should we copy the uncontrolled inner blocks?
```

Comment:
I think we should create a ref as soon as the overlay is edited, similar to how we create a navigation when the uncontrolled inner blocks are edited.

Reply:
Can you help me to understand what you meant here?

The `ref` needs to _match_ the `ref` in the parent Nav block. We're not creating a new one. We're just assigning the _exact_ same one.

Reply:
you ask in the comments what to do with the navigations that don't have a ref.:

> Should we copy the uncontrolled inner blocks?

No, when the user clicks edit overlay on a nav w/ uncontrolled blocks, create the wp_navigation get the ref, asign the ref to the navogation in the post, then copy the ref to the block in the overlay.